src/distilabel/distiset.py

```diff
@@ -193,6 +225,193 @@ def train_test_split(
+                Defaults to None.
+            save_card: Whether to save the dataset card. Defaults to True.
+            save_pipeline_config: Whether to save the pipeline configuration file (aka the `pipeline.yaml` file).
+                Defaults to True.
+            save_pipeline_log: Whether to save the pipeline log file (aka the `pipeline.log` file).
+                Defaults to True.
+
+        Examples:
+            ```python
+            # Save your distiset in a local folder:
+            >>> distiset.save_to_disk(dataset_path="my-distiset")
+            # Save your distiset in a remote storage:
+            >>> storage_options = {
+            ...     "key": os.environ["S3_ACCESS_KEY"],
+            ...     "secret": os.environ["S3_SECRET_KEY"],
+            ...     "client_kwargs": {
+            ...         "endpoint_url": os.environ["S3_ENDPOINT_URL"],
+            ...         "region_name": os.environ["S3_REGION"],
+            ...     },
+            ... }
+            >>> distiset.save_to_disk(dataset_path="my-distiset", storage_options=storage_options)
+            ```
+        """
+        distiset_path = str(distiset_path)
+        for name, dataset in self.items():
+            dataset.save_to_disk(
+                f"{distiset_path}/{name}",
+                max_shard_size=max_shard_size,
+                num_shards=num_shards,
+                num_proc=num_proc,
+                storage_options=storage_options,
+            )
+
+        distiset_config_folder = posixpath.join(distiset_path, DISTISET_CONFIG_FOLDER)
+
+        fs: fsspec.AbstractFileSystem
+        fs, _, _ = fsspec.get_fs_token_paths(
+            distiset_config_folder, storage_options=storage_options
+        )
+        fs.makedirs(distiset_config_folder, exist_ok=True)
+
+        if save_card:
+            # NOTE: Currently the card is not the same if we write to disk or push to the HF hub,
+            # as we aren't generating the README copying/updating the data from the dataset repo.
+            card = self._get_card(repo_id=Path(distiset_path).stem, token=None)
+            new_filename = posixpath.join(distiset_config_folder, "README.md")
+            if storage_options:
+                # Write the card the same way as DatasetCard.save does:
+                with fs.open(new_filename, "w", newline="", encoding="utf-8") as f:
+                    f.write(str(card))
+            else:
+                card.save(new_filename)
+
+        # Write our internal files to the distiset folder by copying them to the distiset folder.
+        if save_pipeline_config and self.pipeline_path:
+            new_filename = posixpath.join(
+                distiset_config_folder, PIPELINE_CONFIG_FILENAME
+            )
+            if self.pipeline_path.exists() and (not fs.isfile(new_filename)):
+                data = yaml.safe_load(
+                    self.pipeline_path.read_text(), Loader=yaml.FullLoader
```

Comment:
I got `TypeError: safe_load() got an unexpected keyword argument 'Loader'`
and fixed it with.
```python
                data = yaml.safe_load(
                    # self.pipeline_path.read_text(), Loader=yaml.FullLoader
                    self.pipeline_path.read_text()
                )
```
I assume for `yaml.safe_load()` one does not need to specify a `Loader` anymore

Reply:
I think I mixed `safe_load` with `load` with the arguments, thanks for the catch!